torch/csrc/distributed/rpc/init.cpp

```diff
@@ -342,9 +342,10 @@ If the future completes with an error, an exception is thrown.
     return RRefContext::getInstance().getDebugInfo();
   });
 
-  module.def("_cleanup_python_rpc_handler", []() {
-    PythonRpcHandler::getInstance().cleanup();
-  });
+  module.def(
+      "_cleanup_python_rpc_handler",
+      []() { PythonRpcHandler::getInstance().cleanup(); },
+      py::call_guard<py::gil_scoped_release>());
```

Comment:
Looks like the GIL will be immediately acquired again after entering `cleanup()`. Is there any specific reason for back-to-back release-acquire here?

Reply:
Oh I see, it is to avoid holding GIL for `PythonRpcHandler::getInstance().`

Reply:
both I and @zhaojuanmao had fallen into this pitfall, so I added intern assert to avoid people falling into again. But this change requires this callsite to release lock.